riff-raff/app/notification/DeployFailureNotifications.scala

```diff
@@ -50,46 +55,90 @@ class DeployFailureNotifications(config: Config,
+      case Some(Right(targets)) => targets
+      case _ => riffRaffTargets
+    }
+  }
+
+  def notifyViaAnghammarad(notificationContents: NotificationContents, targets: List[Target]) = {
+    log.info(s"Sending anghammarad notification with targets: ${targets.toSet}")
+    Anghammarad.notify(
+      subject = notificationContents.subject,
+      message = notificationContents.message,
+      sourceSystem = "riff-raff",
+      channel = All,
+      target = targets,
+      actions = notificationContents.actions,
+      topicArn = anghammaradTopicARN,
+      client = snsClient
+    ).recover { case ex => log.error(s"Failed to send notification (via Anghammarad)", ex) }
+  }
 
-    deriveAnghammaradTargets match {
-      case Right(targets) =>
-        log.info(s"Sending anghammarad notification with targets: ${targets.toSet}")
-        val failureMessage = s"${parameters.deployer.name} for ${parameters.build.projectName} (build ${parameters.build.id}) to stage ${parameters.stage.name} failed."
-        Anghammarad.notify(
-          subject = s"${parameters.deployer.name} failed",
-          message = failureMessage,
-          sourceSystem = "riff-raff",
-          channel = All,
-          target = targets,
-          // TODO: Figure out the correct action for scheduled deploys that don't start
-          actions = uuid.map(id => List(Action("View failed deploy", url(id)))).getOrElse(Nil),
-          topicArn = anghammaradTopicARN,
-          client = snsClient
-        ).recover { case ex => log.error(s"Failed to send notification (via Anghammarad) for $uuid", ex) }
-      case Left(_) =>
-        log.error(s"Failed to derive targets required to notify about failed scheduled deploy: $uuid")
+  def notificationContentsWithTargets(scheduledDeployError: ScheduledDeployError): NotificationContentsWithTargets = {
+    val subject = "Scheduled deploy failed to start"
+    def viewProblematicDeploy(uuid: UUID, status: String) = Action(s"View $status deploy", problematicDeployUrl(uuid))
+    scheduledDeployError match {
+      case SkippedDueToPreviousFailure(record) =>
+        val message = s"A scheduled deploy of ${record.parameters.build.projectName} to ${record.parameters.stage} didn't start because the most recent deploy failed."
+        val redeployAction = Action("Redeploy manually", prefix + routes.DeployController.deployAgainUuid(record.uuid.toString).url)
+        val contents = NotificationContents(subject, message, List(viewProblematicDeploy(record.uuid, "failed"), redeployAction))
+        NotificationContentsWithTargets(contents, getTargets(None, Some(record.parameters)))
+      case SkippedDueToPreviousWaitingDeploy(record) =>
+        val message = s"A scheduled deploy of ${record.parameters.build.projectName} to ${record.parameters.stage} failed to start as a previous deploy was still waiting to be deployed."
+        val contents = NotificationContents(subject, message, List(viewProblematicDeploy(record.uuid, "waiting")))
```

Comment:
Here's an [example](https://riffraff.gutools.co.uk/deployment/view/699d166c-c0e5-4e12-90f0-9f37f89d2190) of this problem. @sihil - any ideas how we should resolve problems like this?